src/services/youtube-api.ts

```diff
@@ -65,74 +97,71 @@ export default class {
       throw new Error('No video found.');
     }
 
-    return this.getVideo(firstVideo.id, shouldSplitChapters);
+    return this.getVideo(firstVideo.url, shouldSplitChapters);
   }
 
   async getVideo(url: string, shouldSplitChapters: boolean): Promise<SongMetadata[]> {
-    const video = await this.cache.wrap(
-      this.youtube.videos.get,
-      cleanUrl(url),
-      {
-        expiresIn: ONE_HOUR_IN_SECONDS,
-      },
-    );
+    const result = await this.getVideosByID([String(getYouTubeID(url))]);
+    const video = result.at(0)!; // TODO undefined handling
 
     return this.getMetadataFromVideo({video, shouldSplitChapters});
   }
 
   async getPlaylist(listId: string, shouldSplitChapters: boolean): Promise<SongMetadata[]> {
-    // YouTube playlist
-    const playlist = await this.cache.wrap(
-      this.youtube.playlists.get,
-      listId,
+    const playlistParams = {
+      searchParams: {
+        part: 'id, snippet, contentDetails',
+        id: listId,
+        key: this.youtubeKey,
+        responseType: 'json',
+      },
+    };
+    const {items: playlists} = await this.cache.wrap(
+      async () => got(YT_BASE_URL + 'playlists', playlistParams).json() as Promise<{items: PlaylistResponse[]}>,
```

Comment:
instead of having to concatenate the base URL and add the API key every time, I would recommend using `got.extend()` in the constructor to create a custom Got instance

Reply:
Created a new commit for this 👍